### ISSUE TITLE
Make nonce part of default Content Security Policy header

### ISSUE THREAD

**Issue by richdougherty** [playframework/playframework#8234]
As [suggested](https://github.com/playframework/playframework/issues/7662#issuecomment-365907951) by @makkarpov:

> Another possibility to fix it is to do something like that:
>
> 1. CSP filter should generate a nonce and attach it as an attribute to the `RequestHeader`. This feature would be also useful when used standalone. When it comes to the CSP policy, it should replace some special token like `@nonce@` to the actual nonce value. This token may be specified as regexp like `'nonce-[^']+'`, so any nonce field in CSP would be transformed to a valid nonce.
>
> 2. Error template should include nonce extracted from the request in all inline content, so it will pass CSP filter.

This would provide a simple way for users and libraries to integrate with Play's CSP header. For example, the play-swagger project (see [issue](https://github.com/iheartradio/play-swagger/issues/174)) could make use of the nonce for inline elements.

**Comment by richdougherty** [comment]
Note: any solution should fix #7662.

**Comment by makkarpov** [comment]
Ok. When I suggested this I wasn't sure whether this approach will be accepted. Now I know that.

I'll try to implement it and create a PR.

**Comment by wsargent** [comment]
I'm running into 

```
[jetty-alpn-agent] Replacing: sun/security/ssl/Alerts
Exception in thread "specs2.fixed.env397465184-3" java.lang.NoClassDefFoundError: sun/security/ssl/SupportedEllipticCurvesExtension
```

I think this is because I'm using latest Java:

```
$ java -version                      
java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)
```

and I don't think the APLN version exists for 161 yet: https://stackoverflow.com/a/41723534/5266

https://www.eclipse.org/jetty/documentation/current/alpn-chapter.html#alpn-versions has 1.8.0u152 as the latest version so I'm going to try that.

**Comment by wsargent** [comment]
@richdougherty got something that passes smoke tests at least, check it out :-)

**Comment by wsargent** [comment]
Fixed via  #8242
